mesop/components/uploader/uploader.py

```diff
@@ -0,0 +1,91 @@
+import io
+from dataclasses import dataclass
+from typing import Any, Callable, Sequence
+
+import mesop.components.uploader.uploader_pb2 as uploader_pb
+from mesop.component_helpers import (
+  insert_component,
+  register_event_handler,
+  register_event_mapper,
+  register_native_component,
+)
+from mesop.events import MesopEvent
+
+
+@dataclass(kw_only=True)
+class UploadedFile:
+  """Uploaded file contents and metadata.
+
+  Attributes:
+      name: File name.
+      size: File size.
+      mime_type: File mime type.
+      contents: File contents.
+  """
+
+  name: str
+  size: int
+  mime_type: str
+  contents: io.BytesIO
+
+
+@dataclass(kw_only=True)
+class UploadEvent(MesopEvent):
```

Comment:
I think it's better to have the API be `file` and just a singular UploadedFile. I know you mentioned supporting multiple file uploads in the future, I think the current API should make single file upload (which IMO will be the common case) as ergonomic as possible.

In the future, if we want to support multiple file uploads, we have a couple options. We could add another attribute here `files` and then `file` would simply be an alias to the first element of `files`. Or we could have two different events (e.g. UploadEvent, UploadMultipleFilesEvent) and emit the different events by overloading the uploader function (e.g. have a parameter like `accept_multiple_files`). I'm thinking the first option isn't bad - having two attributes `file` and `files` on the same event, but we can decide this later.

Reply:
That's great point. I've definitely found some annoyance with how the Javascript api does this too. I've updated to use a file parameter. I agree that using files with an alias to file would be fine when we add multi file uploads.